src/components/ThemeToggle.tsx

```diff
@@ -0,0 +1,37 @@
+import React, {useCallback, useEffect, useState} from 'react';
+import halfMoon from "../assets/half-moon.png"
+import sheep from "../assets/sheep.png"
+
+const ThemeToggle = () => {
+  const [isDark, setIsDark] = useState(false)
+  useEffect(() => {
+    // Checks initial Theme
+    if (document.documentElement.getAttribute("data-theme") == "dark") {
+      console.log("setting")
+      setIsDark(true)
+      localStorage.setItem("theme", "dark")
+    } else {
+      setIsDark(false)
+      localStorage.setItem("theme", "light")
+    }
+  }, [])
+
+  const handleChangeTheme = useCallback(() => {
```

Comment:
The use of `useCallback` seems unnecessary. The callback doesn't have any dependencies and isn't used as a dependency either. I think you can use a normal (arrow) function.